resolver-dns-native-macos/src/main/java/io/netty/resolver/dns/macos/MacOSDnsServerAddressStreamProvider.java

```diff
@@ -72,9 +70,8 @@
     }
 
     private static void loadNativeLibrary() {
-        String name = SystemPropertyUtil.get("os.name").toLowerCase(Locale.UK).trim();
-        if (!name.startsWith("mac")) {
-            throw new IllegalStateException("Only supported on MacOS");
+        if (PlatformDependent.isOsx()) {
```

Comment:
I don't think this is behavioral change because `DnsServerAddressStreamProviders` uses `PlatformDependent.isOsx()` check before trying to load `MacOSDnsServerAddressStreamProvider`.